docs/data/data-grid/tree-data/tree-data.md

```diff
@@ -165,6 +165,38 @@ const invalidRows = [{ path: ['X'] }, { path: ['Y'] }, { path: ['X', 'X'] }];
 
 :::
 
+## Drag-and-drop tree data reordering
+
+With row reordering, users can reorder tree data or move rows from one group to another.
+To enable this feature with row grouping, pass the `rowReordering` prop to the Data Grid component:
+
+```tsx
+<DataGridPro
+  columns={columns}
+  rows={rows}
+  treeData
+  getTreeDataPath={getTreeDataPath}
+  setTreeDataPath={setTreeDataPath}
+  rowReordering
+/>
+```
+
+:::warning
+In order for the cross parent operations to work where there will be a change in the path, you need to pass the `setTreeDataPath()` prop, that works reverse to how [`getTreeDataPath()`](/x/api/data-grid/data-grid-pro/#data-grid-pro-prop-getTreeDataPath) works.
+:::
+
+{{"demo": "TreeDataReordering.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Sync row data with reordered data
+
+If you want to update the external row data, for example to persist it in the local storage, you can use the `onRowOrderChange()` callback and the Data Grid selectors to get the reordered data and sync with the external data.
+
+```tsx
+<DataGridPro onRowOrderChange={handleRowOrderChange} />
```

Comment:
Shouldn't persistance be implemented with the `processRowUpdate` prop?
Same as with editing and row reordering with row grouping?
From what I see in the code, this is already implemented after `setTreeDataPath` returns an updated row.

Reply:
So for row reordering persistence to work with nested data, both `processRowUpdate` and `onRowOrderChange` should be used, correct?

Reply:
Thanks for the details!

> 3. Proper backend data persistence with order tracking

I think this is the scenario we should follow in our persistence demo.
Initially, I thought that the need to use both `processRowUpdate` and `onRowOrderChange` for persistence is weird. But on second thought, that's exactly what you'd do in case of a flat rows structure if you want to support Editing and Row reordering. The difference is that for nested rows, moving the row to a different parent === editing operation, so you have to use Editing API to persist such an action 👍🏻

In our docs we should explain that moving a row from one parent to another edits a row under the hood, and this is why the `processRowUpdate` is necessary for persistance.

Reply:
Updated the demo with a local storage based store that can mimic an actual backend which tracks the order of nested rows in a tree structure: https://deploy-preview-19401--material-ui-x.netlify.app/x/react-data-grid/tree-data/#reordering-persistance

Let me know if it makes sense now?